streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignorTest.java

```diff
@@ -0,0 +1,347 @@
+        // give p1 assignment so already at capacity
+        client.assign(task02, true);
+        // p2 nothing assigned yet
+        createClient(p2, 1);
+        // p3 nothing assigned but has previous standbyTask
+        final ClientState<TaskId> c3 = createClientWithPreviousActiveTasks(p3, 1);
+        c3.addPreviousStandbyTasks(Utils.mkSet(task00));
+
+        final StickyTaskAssignor<Integer> taskAssignor = createTaskAssignor(task01, task00);
+
+        taskAssignor.assign(0);
+
+        assertThat(clients.get(p3).activeTasks(), equalTo(Collections.singleton(task00)));
+        assertThat(clients.get(p2).activeTasks(), equalTo(Collections.singleton(task01)));
+    }
+
+    @Test
+    public void shouldAssignStandbyTasksToClientThatDontHaveSameActiveTask() throws Exception {
+        final TaskId task03 = new TaskId(0, 3);
+        createClientWithActiveTasks(p1, 1, task00);
+        createClientWithActiveTasks(p2, 1, task01);
+        createClientWithActiveTasks(p3, 1, task02);
+        createClientWithActiveTasks(4 , 1, task03);
+
+        final StickyTaskAssignor taskAssignor = createTaskAssignor(task00, task01, task02, task03);
+
+        taskAssignor.assignStandby(1);
+
+        assertThat(clients.get(p1).standbyTasks(), not(hasItems(task00)));
+        assertThat(clients.get(p1).standbyTasks().size(), equalTo(1));
+        assertThat(clients.get(p2).standbyTasks(), not(hasItems(task01)));
+        assertThat(clients.get(p2).standbyTasks().size(), equalTo(1));
+        assertThat(clients.get(p3).standbyTasks(), not(hasItems(task02)));
+        assertThat(clients.get(p3).standbyTasks().size(), equalTo(1));
+        assertThat(clients.get(4).standbyTasks(), not(hasItems(task03)));
+        assertThat(clients.get(4).standbyTasks().size(), equalTo(1));
+    }
+
+    @Test
+    public void shouldAssignMultipleReplicasOfStandbyTask() throws Exception {
+        createClientWithActiveTasks(p1, 1, task00);
+        createClientWithActiveTasks(p2, 1, task01);
+        createClientWithActiveTasks(p3, 1, task02);
+
+        final StickyTaskAssignor taskAssignor = createTaskAssignor(task00, task01, task02);
+        taskAssignor.assignStandby(2);
+
+        assertThat(clients.get(p1).standbyTasks(), equalTo(Utils.mkSet(task01, task02)));
+        assertThat(clients.get(p2).standbyTasks(), equalTo(Utils.mkSet(task02, task00)));
+        assertThat(clients.get(p3).standbyTasks(), equalTo(Utils.mkSet(task00, task01)));
+    }
+
+    @Test
+    public void shouldNotAssignStandbyTaskReplicasWhenAllClientAlreadyHaveTask() throws Exception {
+        createClientWithActiveTasks(p1, 1, task00);
+        final StickyTaskAssignor taskAssignor = createTaskAssignor(task00);
+        taskAssignor.assignStandby(1);
+        assertThat(clients.get(p1).standbyTasks().size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldAssignStandbyToClientWithPreviousActiveTaskWhenDoesntHaveSameActiveTask() throws Exception {
```

Comment:
In the case where the client is already above capacity. The active task could be assigned to another client.